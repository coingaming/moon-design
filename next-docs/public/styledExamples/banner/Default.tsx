import { Banner } from '@heathmont/moon-core';

const Example = () => (
  <Banner title="Title here" description="Lorem ipsum dolor..." />
);

export default Example;
