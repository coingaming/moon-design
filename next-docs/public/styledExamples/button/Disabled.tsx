import { Button } from '@heathmont/moon-core';

const Example = () => <Button disabled>Disabled</Button>;

export default Example;
