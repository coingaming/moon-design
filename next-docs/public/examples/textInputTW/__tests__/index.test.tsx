/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import Bg from '../Bg';
import Default from '../Default';
import Disabled from '../Disabled';
import Error from '../Error';
import Hint from '../Hint';
import Readonly from '../Readonly';
import Sizes from '../Sizes';
import Types from '../Types';
import WithoutLabel from '../WithoutLabel';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

describe('TextInput: type Date', () => {
  it('renders default', () => {
    const testRenderer = renderer.create(renderWithTheme(<Default />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders different sizes', () => {
    const testRenderer = renderer.create(renderWithTheme(<Sizes />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders without label', () => {
    const testRenderer = renderer.create(renderWithTheme(<WithoutLabel />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders with hint', () => {
    const testRenderer = renderer.create(renderWithTheme(<Hint />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders disabled', () => {
    const testRenderer = renderer.create(renderWithTheme(<Disabled />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders error', () => {
    const testRenderer = renderer.create(renderWithTheme(<Error />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('supports background colour', () => {
    const testRenderer = renderer.create(renderWithTheme(<Bg />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('renders readOnly', () => {
    const testRenderer = renderer.create(renderWithTheme(<Readonly />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders different types', () => {
    const testRenderer = renderer.create(renderWithTheme(<Types />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
