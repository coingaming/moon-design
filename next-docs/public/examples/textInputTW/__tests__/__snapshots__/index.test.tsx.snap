// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput: type Date renders default 1`] = `
<div
  className="w-full max-w-full relative z-0 rounded-moon-i-xs"
>
  <input
    className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
    placeholder=""
    type="text"
  />
</div>
`;

exports[`TextInput: type Date renders different sizes 1`] = `
<div
  className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-14"
    >
      Small (sm)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-8 leading-8 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      type="text"
    />
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-16"
    >
      Medium (Default: md)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      type="text"
    />
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-16"
    >
      Large (lg)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-12 leading-[3rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-lg-dt-shared input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      type="text"
    />
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-sm"
  >
    <div
      className="w-full max-w-full relative  bg-gohan rounded-moon-i-sm"
    >
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-transparent h-14 leading-[3.5rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-xl-dt-shared input-xl pt-[1.125rem] input-xl-dt-label input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="Placeholder"
        type="text"
      />
      <label
        className="absolute text-[0.75rem] leading-3 text-trunks top-3 z-[1] transition-all ease-in-out duration-200 left-4"
      >
        Xlarge (xl)
      </label>
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date renders different types 1`] = `
Array [
  <div
    className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
  >
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Number
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-number-clear input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="e.g. 12345"
        type="number"
      />
    </div>
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Date
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-d input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder=""
        type="date"
      />
    </div>
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Time
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-t input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder=""
        type="time"
      />
    </div>
  </div>,
  <div
    className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
  >
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Datetime local
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-d input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder=""
        type="datetime-local"
      />
    </div>
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Email
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="e.g. john.doe@domain.com"
        type="email"
      />
    </div>
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Password
      </label>
      <div
        className="w-full max-w-full relative rounded-moon-i-xs"
      >
        <input
          className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
          placeholder="Password"
          type="password"
        />
        <div
          className="text-moon-14 text-bulma absolute top-2/4 mt-[-0.75rem] z-[3] underline cursor-pointer right-4"
          onClick={[Function]}
          role="alert"
        >
          Show
        </div>
      </div>
    </div>
  </div>,
  <div
    className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
  >
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Search
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="e.g. Search something"
        type="search"
      />
    </div>
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Tel
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="e.g. +372 123 4567"
        type="tel"
      />
    </div>
    <div
      className="w-full max-w-full relative z-0 rounded-moon-i-xs"
    >
      <label
        className="block text-bulma pb-2 text-moon-16"
      >
        Url
      </label>
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="e.g. https://domain.com"
        type="url"
      />
    </div>
  </div>,
]
`;

exports[`TextInput: type Date renders disabled 1`] = `
<div
  className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
>
  <div
    className="w-full max-w-full relative z-0 opacity-30 cursor-not-allowed rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-16 opacity-30 cursor-not-allowed"
    >
      Medium (md)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      disabled={true}
      placeholder="Placeholder"
      type="text"
    />
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
  <div
    className="w-full max-w-full relative z-0 opacity-30 cursor-not-allowed rounded-moon-i-sm"
  >
    <div
      className="w-full max-w-full relative  bg-gohan rounded-moon-i-sm"
    >
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-transparent h-14 leading-[3.5rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-xl-dt-shared input-xl pt-[1.125rem] input-xl-dt-label input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        disabled={true}
        placeholder="Placeholder"
        type="text"
      />
      <label
        className="absolute text-[0.75rem] leading-3 text-trunks top-3 z-[1] transition-all ease-in-out duration-200 left-4"
      >
        Xlarge (xl)
      </label>
    </div>
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
</div>
`;

exports[`TextInput: type Date renders error 1`] = `
<div
  className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-16"
    >
      Medium (md)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none shadow-input-err hover:shadow-input-err focus:shadow-input-err bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      type="text"
    />
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-chiChi"
      role="alert"
    >
      Error message
    </p>
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-sm"
  >
    <div
      className="w-full max-w-full relative  bg-gohan rounded-moon-i-sm"
    >
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none shadow-input-err hover:shadow-input-err focus:shadow-input-err bg-transparent h-14 leading-[3.5rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-xl-dt-shared input-xl pt-[1.125rem] input-xl-dt-label input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="Placeholder"
        type="text"
      />
      <label
        className="absolute text-[0.75rem] leading-3 text-trunks top-3 z-[1] transition-all ease-in-out duration-200 left-4"
      >
        Xlarge (xl)
      </label>
    </div>
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-chiChi"
      role="alert"
    >
      Error message
    </p>
  </div>
</div>
`;

exports[`TextInput: type Date renders readOnly 1`] = `
<div
  className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-16"
    >
      Medium (md)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      readOnly={true}
      type="text"
      value="Read only text"
    />
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-sm"
  >
    <div
      className="w-full max-w-full relative  bg-gohan rounded-moon-i-sm"
    >
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-transparent h-14 leading-[3.5rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-xl-dt-shared input-xl pt-[1.125rem] input-xl-dt-label input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="Placeholder"
        readOnly={true}
        type="text"
        value="Read only text"
      />
      <label
        className="absolute text-[0.75rem] leading-3 text-trunks top-3 z-[1] transition-all ease-in-out duration-200 left-4"
      >
        Xlarge (xl)
      </label>
    </div>
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
</div>
`;

exports[`TextInput: type Date renders with hint 1`] = `
<div
  className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-16"
    >
      Medium (md)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      type="text"
    />
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-sm"
  >
    <div
      className="w-full max-w-full relative  bg-gohan rounded-moon-i-sm"
    >
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-transparent h-14 leading-[3.5rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-xl-dt-shared input-xl pt-[1.125rem] input-xl-dt-label input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="Placeholder"
        type="text"
      />
      <label
        className="absolute text-[0.75rem] leading-3 text-trunks top-3 z-[1] transition-all ease-in-out duration-200 left-4"
      >
        Xlarge (xl)
      </label>
    </div>
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
</div>
`;

exports[`TextInput: type Date renders without label 1`] = `
<div
  className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-gohan h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      type="text"
    />
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-sm"
  >
    <div
      className="w-full max-w-full relative  bg-gohan rounded-moon-i-sm"
    >
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-transparent h-14 leading-[3.5rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-xl-dt-shared input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="Placeholder"
        type="text"
      />
      <label
        className="absolute text-[0.75rem] leading-3 text-trunks top-3 z-[1] transition-all ease-in-out duration-200 left-4"
      />
    </div>
  </div>
</div>
`;

exports[`TextInput: type Date supports background colour 1`] = `
<div
  className="flex flex-col lg:flex-row justify-around items-end w-full gap-2"
>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-xs"
  >
    <label
      className="block text-bulma pb-2 text-moon-16"
    >
      Medium (md)
    </label>
    <input
      className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-beerus h-10 leading-10 rounded-moon-i-xs hover:rounded-moon-i-xs focus:rounded-moon-i-xs invalid:rounded-moon-i-xs before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
      placeholder="Placeholder"
      type="text"
    />
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
  <div
    className="w-full max-w-full relative z-0 rounded-moon-i-sm"
  >
    <div
      className="w-full max-w-full relative  bg-beerus rounded-moon-i-sm"
    >
      <input
        className="block w-full max-w-full py-0 px-4 m-0 appearance-none text-[1rem] text-bulma transition-shadow box-border relative z-[2] shadow-input hover:shadow-input-hov focus:shadow-input-focus focus:outline-none bg-transparent h-14 leading-[3.5rem] rounded-moon-i-sm hover:rounded-moon-i-sm focus:rounded-moon-i-sm invalid:rounded-moon-i-sm before:box-border after:box-border placeholder:text-trunks placeholder:opacity-100 placeholder:transition-opacity placeholder:delay-75 read-only:outline-0 read-only:border-none read-only:cursor-not-allowed read-only:hover:shadow-input read-only:focus:shadow-input input-xl-dt-shared input-xl pt-[1.125rem] input-xl-dt-label input-dt-shared invalid:shadow-input-err invalid:hover:shadow-input-err invalid:focus:shadow-input-err"
        placeholder="Placeholder"
        type="text"
      />
      <label
        className="absolute text-[0.75rem] leading-3 text-trunks top-3 z-[1] transition-all ease-in-out duration-200 left-4"
      >
        Xlarge (xl)
      </label>
    </div>
    <p
      className="inline-block mt-2 ps-4 text-moon-12 text-trunks"
      role="alert"
    >
      Informative message holder
    </p>
  </div>
</div>
`;
